### PR TITLE
[SQL] Recognize ROW_NUMBER == 1 as TopK

### LINKED ISSUE #1184: OVER only supports ascending sorting
1. SQL
```
CREATE VIEW TOP_EVENT_DURATIONS_V AS
SELECT duration
,      event_type_id
FROM   (SELECT duration
        ,      event_type_id
        ,      ROW_NUMBER() OVER (PARTITION BY event_type_id
                                  ORDER BY duration DESC) AS rnum
        FROM   EVENT_DURATION_V)
WHERE   rnum = 1
ORDER BY 1 DESC
;
```

2. Error
```
Program status: {'SqlError': [{'startLineNumber': 0, 'startColumn': 0, 'endLineNumber': 0, 'endColumn': 0, 'warning': False, 'errorType': 'Not yet implemented', 'message': 'OVER only supports ascending sorting SELECT `DURATION`, `EVENT_TYPE_ID`, `SITE_ID`, ROW_NUMBER() OVER (PARTITION BY `EVENT_TYPE_ID` ORDER BY `DURATION` DESC) AS `w0$o0`\nFROM `schema`.`EVENT_DURATION_V`'}]}
```

### LINKED ISSUE #1185: ROW_NUMBER raises a "Not yet implemented" error
1. SQL
```
CREATE VIEW TOP_EVENT_DURATIONS_V AS
SELECT duration
,      site_id
FROM   (SELECT duration
        ,      site_id
        ,      ROW_NUMBER() OVER (PARTITION BY site_id
                                  ORDER BY duration ASC) AS rnum
        FROM   EVENT_DURATION_V)
WHERE   rnum = 1
;
```

2. Error
```
Program status: {'SqlError': [{'startLineNumber': 0, 'startColumn': 0, 'endLineNumber': 0, 'endColumn': 0, 'warning': False, 'errorType': 'Not yet implemented', 'message': 'ROW_NUMBER'}]}
```

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

Fixes #1184 
Fixes #1185

### CODE REVIEW

**Comment by mihaibudiu** [comment]
This fails in CI due to unrelated issues

**Comment by lalithsuresh** [comment]
I've rerun the workflow. There is an odd race condition in the test that fails.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TopKTests.java [hunk 1]
```diff
@@ -114,6 +114,73 @@ public void issue1174() {
...
+                "event_clear_date BIGINT -- epoch\n" +
+                ");\n" +
+                "\n" +
+                "CREATE VIEW EVENT_DURATION_V AS\n" +
+                "SELECT (event_date - event_clear_date) AS duration\n" +
+                ",      event_type_id\n" +
+                ",      site_id\n" +
+                "FROM   event_t\n" +
+                "WHERE  event_clear_date IS NOT NULL\n" +
+                ";\n" +
```
Comment (lalithsuresh) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/TopKTests.java:133]: Use a multi-line string for these two tests

**Review by lalithsuresh — APPROVED** [review]

**Comment by mihaibudiu** [comment]
@abhizer from now on we should use Java text blocks for tests, they look much better. Refactored all tests.

**Comment by abhizer** [comment]
Ah nice, I didn't know we could do that, that's neat!

**Pull request merged**